exporters/trace/jaeger/uploader.go

```diff
@@ -38,24 +38,40 @@ type batchUploader interface {
 type EndpointOption func() (batchUploader, error)
 
 // WithAgentEndpoint instructs exporter to send spans to jaeger-agent at this address.
-// For example, localhost:6831.
-func WithAgentEndpoint(agentEndpoint string, options ...AgentEndpointOption) EndpointOption {
+// Agent endpoint is resolved in the following order: options, env vars, and then default values
+// Default agent endpoint value: localhost:6832
+func WithAgentEndpoint(options ...AgentEndpointOption) EndpointOption {
 	return func() (batchUploader, error) {
-		if agentEndpoint == "" {
-			return nil, errors.New("agentEndpoint must not be empty")
-		}
-
 		o := &AgentEndpointOptions{
 			agentClientUDPParams{
-				HostPort:            agentEndpoint,
```

Comment:
I would recommend consolidating and focusing the built functionality in this PR.

1. I would get rid of `agentEndpointFromEnv` and replace it with a "`envOr`" type function that generically returns an environment variable value if it exists or a default if not.
  E.g.
  ```
func envOr(key, default string) string {
	if v, ok := os.LookupEnv(key); ok {
		return v
	}
	return default
}
```
2. Use the `envOr` func here to set the defaults/environment values that would then be overriden by explicit options passed.
  E.g.
  ```
agentClientUDPParams{
	Host: envOr(envAgentHost, "localhost"),
	Port: envOr(envAgentPort, "6832"),
	...
}